File: packages/expo-scroll-passthrough/ios/ExpoScrollPassthroughModule.swift
import ExpoModulesCore

public class ExpoScrollPassthroughModule: Module {
  // Each module class must implement the definition function. The definition consists of components
  // that describes the module's functionality and behavior.
  // See https://docs.expo.dev/modules/module-api for more details about available components.
  public func definition() -> ModuleDefinition {
    // Sets the name of the module that JavaScript code will use to refer to the module. Takes a string as an argument.
    // Can be inferred from module's class name, but it's recommended to set it explicitly for clarity.
    // The module will be accessible from `requireNativeModule('ExpoScrollPassthrough')` in JavaScript.
    Name("ExpoScrollPassthrough")

    // Enables the module to be used as a native view. Definition components that are accepted as part of the
    // view definition: Prop, Events.
    View(ExpoScrollPassthroughView.self) {
      // Defines a setter for the `url` prop.
      Prop("url") { (view: ExpoScrollPassthroughView, url: URL) in
        if view.webView.url != url {
          view.webView.load(URLRequest(url: url))
        }
      }

      Events("onLoad")
    }
  }
}

File: packages/expo-scroll-passthrough/android/src/main/java/expo/modules/scrollpassthrough/ExpoScrollPassthroughModule.kt
package expo.modules.scrollpassthrough

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.net.URL

class ExpoScrollPassthroughModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoScrollPassthrough")

    View(ExpoScrollPassthroughView::class) {}
    View(ExpoScrollPassthroughView2::class) {}
    View(ExpoScrollPassthroughView3::class) {}
  }
}
